openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -274,6 +274,27 @@ def get_from_id(cls, course_id):
 
         return course_overview or cls.load_from_module_store(course_id)
 
+    @classmethod
+    def get_from_ids_if_exists(cls, course_ids):
+        """
+        Return a dict mapping course_ids to CourseOverviews, if they exist.
+
+        This method will *not* generate new CourseOverviews or delete outdated
+        ones. It exists only as a small optimization used when CourseOverviews
+        are known to exist, for common situations like the student dashboard.
+
+        Callers should assume that this list is incomplete and fall back to
+        get_from_id if they need to guarantee CourseOverview generation.
+        """
+        return {
+            overview.id: overview
+            for overview
+            in cls.objects.select_related('image_set').filter(
```

Comment:
Thank you.  Very clear.  You can see how much Django my safe-templates and Ruby work has gotten me.  I was conflating the select_related and the filtering.

To make sure I'm completely clear, in the unlikely case that you mentioned of a `CourseOverview` that doesn't yet have `thumbnails`, the `CourseOverview` could still be found and returned.  The `select_related` is not part of the filtering.  Is this correct?